docs/custom-images-contract.rst

```diff
@@ -0,0 +1,113 @@
+.. _custom-images-contract:
+
+======================
+Custom Images contract
+======================
+
+The contract is a set of requirements for crawler custom docker images to comply running with Scrapy Cloud.
+
+The contract is already implemented in Python and is known as :ref:`scrapinghub-entrypoint-scrapy <sh-entrypoint-scrapy>`. If you don’t use this package with your crawler image, you should implement Custom Images contract by yourself, or your image will not work properly in Scrapy Cloud.
```

Comment:
In a word, there's only one Custom image contract, and `scrapinghub-entrypoint-scrapy` is an implementation of this contract with a lot of additional cool features, and it doesn't have 'competitors' so far. But our platform should stay flexible enough to grow in different directions. I think someone may use its own implementation in the following scenarios:
- a need of additional features special for the project
- drop/replace existing features with a different ones
- write non-python crawlers using any other scraping framework ([example](https://github.com/dangra/anemone-spider))

Probably @dangra or @chekunkov can extend my thoughts, and we can add it to this doc.